Source/ProgressionSystemRuntime/Public/Data/PSWorldSubsystem.h

```diff
@@ -170,8 +178,16 @@ class PROGRESSIONSYSTEMRUNTIME_API UPSWorldSubsystem : public UWorldSubsystem
 	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
 	void OnGameStateChanged(ECurrentGameState CurrentGameState);
 
+	/** Updates the chosen player mesh on the level and switch current save game row */
+	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
+	void ChangeCurrentPlayer(UPSSpotComponent* SpotComponent);
+	
 	/** Checks if the current character is unlocked and the player is allowed to play, and if not allowed, sets the previous character. */
 	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
 	void CheckAndSetCharacterUnlockStatus();
+
+	/** Switches back to the last spot character in case of player tried to play with locked character as client */
+	UFUNCTION(BlueprintCallable, Category = "C++", meta = (BlueprintProtected))
+	void SwitchToLastSpotCharacter();
 	
 };
```

Comment:
Per [d179119](https://github.com/JanSeliv/Bomber/commit/d179119c6dad03ab3bb0d081c26ae1714d392b94), this bug looks a bit different now. Please fix it in the Progression System logic.

1. Reset progression so that only the first Bastet character is unlocked.
2. Switch LEFT from Bastet to the last character, which is the locked Fori.
3. Reproduce the bug: start the game as a different player, then go back to the menu.
4. Notice that the player is sent back to Roger, which is also locked, instead of Bastet.

**Bug:** When switching from Fori (the 4th locked character), the player is returned to Roger (the 3rd locked character). 
See video: [link](https://github.com/user-attachments/assets/4e8e2a89-6749-4ddb-9dcb-5b77527dacf1)

**Expected result:** The player should be returned to the last active character, which is Bastet (the 1st unlocked character).

This happens because the `SetActive` function in the `ChangeSpotVisibilityStatus` is only called for the current character.  
If you run `DisplayAll MySkeletalMeshComponent bIsActive`, you'll see that all other locked characters are still active. This is why the NMM module sends the player back to Roger - because it stays active while it had to be deactivated by PS module.

**How to fix:**  
Instead of deactivating just the current character, you should call `SetActive` for **all** character meshes when updating the progression. Additionally, activate the current spot's mesh when unlocking a new progression level.

Reply:
solved as part of  https://github.com/h4rdmol/ProgressionSystem/pull/3/commits/eded2e446ff1204eb63d3800bbcce887c41fa82a